src/main/java/com/iota/iri/network/pipeline/BroadcastStage.java

```diff
@@ -0,0 +1,59 @@
+package com.iota.iri.network.pipeline;
+
+import com.iota.iri.controllers.TransactionViewModel;
+import com.iota.iri.network.NeighborRouter;
+import com.iota.iri.network.neighbor.Neighbor;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Map;
+import java.util.Optional;
+
+/**
+ * The {@link BroadcastStage} takes care of broadcasting newly received transactions to all neighbors except the
+ * neighbor from which the transaction originated from.
+ */
+public class BroadcastStage {
+
+    private static final Logger log = LoggerFactory.getLogger(BroadcastStage.class);
+
+    private NeighborRouter neighborRouter;
+
+    /**
+     * Creates a new {@link BroadcastStage}.
+     * 
+     * @param neighborRouter The {@link NeighborRouter} instance to use to broadcast
+     */
+    public BroadcastStage(NeighborRouter neighborRouter) {
+        this.neighborRouter = neighborRouter;
+    }
+
+    /**
+     * Extracts the transaction and then broadcasts it to all neighbors. If the transaction originated from a neighbor,
+     * it is not sent to that given neighbor.
+     * 
+     * @param ctx the broadcast stage {@link ProcessingContext}
+     * @return the same ctx as passed in
+     */
+    public ProcessingContext process(ProcessingContext ctx) {
+        BroadcastPayload payload = (BroadcastPayload) ctx.getPayload();
```

Comment:
Casts are usually a sign of a smell in java. Better find a way to get rid of them if possible.

Reply:
see above